vision/conv_mnist/conv_mnist.jl

```diff
@@ -1,240 +1,218 @@
-                BSON.@save modelpath model epoch
-            end
-            @info "Model saved in \"$(modelpath)\""
-        end
-    end
-end
+#===== ARRAY SIZES =====#
 
-# The function `train` performs the following tasks:
+# A layer like Conv((5, 5), 1=>6) takes 5x5 patches of an image, and matches them to each
+# of 6 different 5x5 filters, placed at every possible position. These filters are here:
 
-# * Checks whether there is a GPU available and uses it for training the model. Otherwise, it uses the CPU.
-# * Loads the MNIST data using the function `get_data`.
-# * Creates the model and uses the [ADAM optimiser](https://fluxml.ai/Flux.jl/stable/training/optimisers/#Flux.Optimise.ADAM) with weight decay.
-# * Loads the [TensorBoardLogger.jl](https://github.com/JuliaLogging/TensorBoardLogger.jl) for logging data to Tensorboard.
-# * Creates the function `report` for computing the loss and accuracy during the training loop. It outputs these values to the TensorBoardLogger.
-# * Runs the training loop using [Flux’s training routine](https://fluxml.ai/Flux.jl/stable/training/training/#Training). For each epoch (step), it executes the following:
-#   * Computes the model’s predictions.
-#   * Computes the loss.
-#   * Updates the model’s parameters.
-#   * Saves the model `model.bson` every `checktime` epochs (defined as argument above.)
+Conv((5, 5), 1=>6).weight |> summary  # 5×5×1×6 Array{Float32, 4}
 
-# ## Run the example 
+# This layer can accept any size of image; let's trace the sizes with the actual input:
 
-# We call the  function `train`:
+#=
 
-if abspath(PROGRAM_FILE) == @__FILE__
-    train()
-end
+julia> x1 |> size
+(28, 28, 1, 64)
+
+julia> lenet[1](x1) |> size  # after Conv((5, 5), 1=>6, relu),
+(24, 24, 6, 64)
+
+julia> lenet[1:2](x1) |> size  # after MaxPool((2, 2))
+(12, 12, 6, 64)
+
+julia> lenet[1:3](x1) |> size  # after Conv((5, 5), 6 => 16, relu)
+(8, 8, 16, 64)
+
+julia> lenet[1:4](x1) |> size  # after MaxPool((2, 2))
+(4, 4, 16, 64)
+
+julia> lenet[1:5](x1) |> size  # after Flux.flatten 
+(256, 64)
+
+=#
+
+# Flux.flatten is just reshape, preserving the batch dimesion (64) while combining others (4*4*16).
+# This 256 must match the Dense(256 => 120). Here is how to automate this, with Flux.outputsize:
+
+lenet2 = Flux.@autosize (28, 28, 1, 1) Chain(
+    Conv((5, 5), 1=>6, relu),
+    MaxPool((2, 2)),
+    Conv((5, 5), _=>16, relu),
+    MaxPool((2, 2)),
+    Flux.flatten,
+    Dense(_ => 120, relu),
+    Dense(_ => 84, relu), 
+    Dense(_ => 10),
+)
+
+#===== LOADING =====#
+
+# During training, the code above saves the model state to disk. Load the last version:
+
+loaded_state = JLD2.load(filename, "lenet_state");
+
+# Now you would normally re-create the model, and copy all parameters into that.
+# We can use lenet2 from just above:
+
+Flux.loadmodel!(lenet2, loaded_state)
+
+# Check that it now agrees with the earlier, trained, model:
+
+lenet2(cpu(x1)) ≈ cpu(lenet(x1))
```

Comment:
Just repeating the suggestion in https://github.com/FluxML/model-zoo/pull/397#discussion_r1185650292 to print some of these out so that users running the script directly aren't completely left without feedback.

Reply:
See what you think of [9bf6c63](https://github.com/FluxML/model-zoo/pull/394/commits/9bf6c636917dc9913abaf6c56c31c4244dfd8dd3), adds more `@show`.

It doesn't print many headings etc. when run as a script. Could add more but it's all extra clutter. If you run this as a script, I think you need the source open in another window to learn anything anyway.